pkg/snapstore/s3_snapstore.go

```diff
@@ -141,8 +143,10 @@ func readAWSCredentialsJSONFile(filename string) (session.Options, error) {
 
 	return session.Options{
 		Config: aws.Config{
-			Credentials: credentials.NewStaticCredentials(awsConfig.AccessKeyID, awsConfig.SecretAccessKey, ""),
-			Region:      pointer.StringPtr(awsConfig.Region),
+			Credentials:      credentials.NewStaticCredentials(awsConfig.AccessKeyID, awsConfig.SecretAccessKey, ""),
+			Region:           pointer.StringPtr(awsConfig.Region),
+			Endpoint:         awsConfig.Endpoint,
```

Comment:
What happen when there is no `Endpoint` configure as `Endpoint` used only for S3 compatible providers? Have you tested that scenario?

Reply:
Nothing happens, the Endpoint just remains `nil`. This is the same as not setting it at all, like we do today https://github.com/gardener/etcd-backup-restore/blob/c48c9973106837a87a652da131cbd6d5bf19b32a/pkg/snapstore/s3_snapstore.go#L78.